runner/websocket.go

```diff
@@ -143,8 +143,17 @@ func connect(ctx context.Context, server *state.Server, globalCollectionOpts sta
 				server.Grant.Store(&grant)
 			} else if message.GetPause() != nil {
 				server.Pause.Store(message.GetPause().Pause)
-			} else if message.GetExplainRun() != nil {
-				logger.PrintVerbose("ExplainRun: %v", message.GetExplainRun()) // TODO
+			} else if message.GetQueryRun() != nil {
+				q := message.GetQueryRun()
+				logger.PrintVerbose("Query run %d received: %s", q.Id, q.QueryText)
+				server.QueryRunsMutex.Lock()
+				server.QueryRuns = append(server.QueryRuns, state.QueryRun{
```

Comment:
As discussed during today's meeting, this should ignore duplicate requests (based on the ID) in case the server has a retry mechanism in the future. Maybe this should be a map instead of an array?